text/0000-build-std/5-stage-1b.md

```diff
@@ -0,0 +1,986 @@
+
+This is not desirable as it is unclear that supporting `path`/`git` sources
+which shadow standard library crates was a deliberate choice and so enabling
+that pattern to be used more widely when not necessary is needlessly permissive.
+
+In addition, when combined with a `git`/`path` source, the `version` constraint
+also applies to package from the `git`/`path` source. If `version` were used
+alongside `builtin`, then this behaviour would be a poor fit as..
+
+- ..the `std`, `alloc` and `core` crates all currently have a version of `0.0.0`
+
+- ..choosing different version requirements for different `builtin` crates has
+  no purpose
+
+↩ [*Proposal*][proposal]
+
+## Why disallow builtin dependencies on other crates?
+[rationale-no-builtin-other-crates]: #why-disallow-builtin-dependencies-on-other-crates
+
+`builtin` dependencies could be accepted on two other crates - dependencies of
+the standard library, like `compiler_builtins`, or other crates in the sysroot
+added manually by users, however:
+
+- The standard library's dependencies are not part of the stable interface of
+  the standard library and it is not desirable that users can observe their
+  existence or depend on them directly
+
+- Other crates in the sysroot added by users are not something that can
+  reasonably be supported by build-std and these crates should become regular
+  dependencies
+
+↩ [*Proposal*][proposal]
+
+## Why allow all names for `builtin` crates on nightly?
+[rationale-nightly-builtin-crates]: #why-allow-all-names-for-builtin-crates-on-nightly
+
+For any crate shipped with the standard library in the sysroot, the user can
+already write an `extern crate` declaration to use it. All crates other than
+`std`, `alloc` or `core` are marked unstable either explicitly or implicitly
+with the use of `-Zforce-unstable-if-unmarked` so this does not allow items from
+these crates to be used on stable.
+
+For example, some users write benchmarks using `libtest` and have written
+`extern crate test` without the `#[cfg(test)]` attribute to load the crate.
+There may be other niche uses of unstable sysroot crates that this enables to
+continue on nightly toolchains.
+
+All names are permitted for `builtin` crates rather than an allowlist to avoid
+Cargo needing to hardcode the names of many of the crates in the sysroot, which
+are inherently unstable.
+
+↩ [*Proposal*][proposal]
+
+## Why not migrate to always requiring explicit standard library dependencies?
+[rationale-no-migration]: #why-not-migrate-to-always-requiring-explicit-standard-library-dependencies
+
+Explicit standard library dependencies with `builtin = true` will necessarily
+only be understood by newer versions of Cargo.
+
+If all packages were required to add explicit dependencies (perhaps over an
+edition or through some other mechanism), then every crate would require the
+newest version of Cargo to be understood, effectively raising the MSRV of every
+Rust crate.
```

Comment:
If its over an edition, then thats already the case and not a problem.

Reply:
I feel I'm missing something with this as it doesn't make sense to me.  Cargo would always be supporting old editions so if we require explicit std deps, then that would only be for packages on that edition.  There wouldn't be an edition to opt-in to say "break if my deps are on an old edition".

Nor does the edit make sense.